### ISSUE TITLE
[beta] Navigating to different challenges repeatedly should "reset" code editor's undo history

### ISSUE THREAD

**Issue by ivantedja** [freeCodeCamp/freeCodeCamp#10972]
Steps to reproduce:
1. go to one of a challenge
2. click "Map"
3. click another challenge
4. repeat step 2 and 4 as many as you want
5. CTRL-Z (undo) in the editor will restore the editor to previous challenge

Expectation:
- when a challenge is opened, the CTRL-Z (undo) history expected to be cleared

Related with [this](https://github.com/FreeCodeCamp/FreeCodeCamp/issues/806#issuecomment-249366799)

Codemirror has ability to clear undo history but in beta site we are using React (`react-codemirror`)

So the way to handle it might be different

**Comment by systimotic** [comment]
I tried solving this issue, but I came to the conclusion I don't yet have the skill to do this. 😞 I'm new to the FreeCodeCamp project and React. This went above my head. In order to hopefully save the next person some time, here are my findings. 👇 
- Going from @ivantedja's findings in #806, I managed to create some code that cleared the undo history. This works by setting a `ref` on the component, and using `this.refs.editor.getCodeMirror().doc.clearHistory();`. So far so good! 😅 
- Although the previous part was a challenge, this is where I gave up. The clearing needs to happen at the right time. After the content is changed, but before the user starts interacting. One caveat of the clearing code is that it only works _after_ the component's `componentDidMount()` (source: [react-codemirror docs](https://github.com/JedWatson/react-codemirror#methods)). I could not figure out how to make this work.
[This is the closest I got](https://github.com/systimotic/FreeCodeCamp/commit/32a34837aa56f97bfbc0cc3f1abba2ed50ae0130). If somebody ends up fixing this, I'd love to hear. I'd really like to see what I should've done to fix this. Good luck! 🙌
